url47ebb27197_asirbu_CorrectiveEvolution_tgz-pJ8-eu_fbk_soa_evolution_engine_impl_test_ProblemToSMVFuncTestJ8/src/eu/fbk/soa/evolution/sts/minimizer/TraceEquivalenceMinimizer.java

```diff
@@ -0,0 +1,57 @@
+package eu.fbk.soa.evolution.sts.minimizer;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import org.apache.log4j.Logger;
+
+import eu.fbk.soa.evolution.sts.STS;
+import eu.fbk.soa.util.ConfigUtils;
+
+
+public class TraceEquivalenceMinimizer implements STSMinimizer {
+
+	static Logger logger = Logger.getLogger(TraceEquivalenceMinimizer.class);
+		
+	@Override
+	public void minimizeSTS(STS inputSTS, File outputFile) {
+		String input2Dot = inputSTS.toDot();
+		
+		try {
+			File tempFile = File.createTempFile("temp", "dot");
+//			tempFile = new File("temp.dot");
+			FileWriter fout = new FileWriter(tempFile);
+	        fout.write(input2Dot);
+	        fout.flush();
+	        fout.close();
+	        this.minimizeSTS(tempFile, outputFile);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		
+	}
+	
+	
+	@Override
+	public void minimizeSTS(File inputFile, File outputFile) {
+		String mcrl2Path;
+		try {
+			mcrl2Path = ConfigUtils.getProperty("mcrl2Path");
+		
+			String[] cmdarray = {mcrl2Path + "ltsconvert.exe", "-etrace", 
+					inputFile.getPath(), outputFile.getPath()};
+			ProcessBuilder builder = new ProcessBuilder(cmdarray);
+			builder.redirectErrorStream(true);
+			
+			Process proc = new ProcessBuilder(cmdarray).start();
+	        proc.waitFor();
+	        
+		} catch (IOException e) {
+			e.printStackTrace();
+		} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)